process_report/process_report.py

```diff
@@ -112,21 +172,99 @@ def remove_billables(dataframe, pi, projects, output_file):
 
     So this *keeps* the projects/pis that should not be billed.
     """
-    filtered_dataframe = dataframe[dataframe['Manager (PI)'].isin(pi) | dataframe['Project - Allocation'].isin(projects)]
+    filtered_dataframe = dataframe[dataframe[PI_FIELD].isin(pi) | dataframe[PROJECT_FIELD].isin(projects)]
     filtered_dataframe.to_csv(output_file, index=False)
 
+
 def export_pi_billables(dataframe: pandas.DataFrame, output_folder):
     if not os.path.exists(output_folder):
         os.mkdir(output_folder)
 
-    invoice_month = dataframe['Invoice Month'].iat[0]
-    pi_list = dataframe['Manager (PI)'].unique()
+    invoice_month = dataframe[INVOICE_DATE_FIELD].iat[0]
+    pi_list = dataframe[PI_FIELD].unique()
 
     for pi in pi_list:
-        pi_projects = dataframe[dataframe['Manager (PI)'] == pi]
-        pi_instituition = pi_projects['Institution'].iat[0]
+        if pandas.isna(pi): 
+            continue
+        pi_projects = dataframe[dataframe[PI_FIELD] == pi]
+        pi_instituition = pi_projects[INSTITUTION_FIELD].iat[0]
         pi_projects.to_csv(output_folder + f"/{pi_instituition}_{pi}_{invoice_month}.csv")
-        
+
+
+def apply_credits_new_pi(dataframe, old_pi_file):
+    """Applies the New PI Credit. This credit function expects the 
+    command arguement `old-pi-file` to be set, pointing to a txt file containing old PIs"""
+    new_pi_credit_code = "0002"
+    new_pi_credit_amount = 1000
+
+    old_pi_list = set()
+
+    try:
+        with open(old_pi_file) as f:
+            for pi in f: 
+                old_pi_list.add(pi.strip())
+
+    except FileNotFoundError:
+        print("Applying credit 0002 failed. Old PI file does not exist")
+        sys.exit(1)
+
+    pi_list = dataframe[PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pandas.isna(pi): 
+            continue   # NaN check
```

Comment:
You should at the very least log a scary warning here, by now we should have removed the non-billable projects and therefore if there's something without a PI it should be logged because it's not expected.

Reply:
@knikolla I can make a function that validates the dataframe after we remove non-billable projects. For now, I'll only check if the PI name field is empty and log it. Is that fine for now?